docs/user-guide.rst

```diff
@@ -757,6 +757,30 @@ make sure you don't define any volumes with the name ``docker``.
       size: 16Gi
       type: ssd
 
+Feature Gates
+~~~~~~~~~~~~~
+
+Feature gates can be enabled or disabled on Kubernetes components through the
+Tarmak configuration file. The feature gates can be set on the API server,
+Kubelet, Scheduler, Controller Manager and Kube-Proxy that will take effect cluster
+wide. To change feature gates, each component takes a list of stings that
+will be applied to their corresponding command line flags or configuration
+file under the Kubernetes code block like following:
+
+.. code-block:: yaml
+
+    kubernetes:
+      featureGates:
```

Comment:
@JoshVanL I would like to see the same config style as we already have for admission controller, public elb. 
This would mean we do `kubernetes -> apiServer -> featureGates` instead of `kubernetes -> featureGates -> apiServer`.

A full example:

```
  kubernetes:
    apiServer:
      featureGates:
        - "AllAlpha=true"
        - "APIResponseCompression=false"
    kubelet:
      featureGates:
        - "CustomPodDNS=true"
    kubeProxy:
      featureGates:
        - "CSIPersistentVolume=true"
    controllerManager:
    scheduler:
```


In a full example this would become:

```
  kubernetes:
    apiServer:
      featureGates:
        - "AllAlpha=true"
        - "APIResponseCompression=false"
      enableAdmissionControllers:
        - "DefaultStorageClass"
      disableAdmissionControllers:
        - "DefaultTolerationSeconds"
      public: true
```

This will remove a lot of the confusion.

Reply:
:+1: